bundles/org.eclipse.ui.editors/src/org/eclipse/ui/texteditor/MarkerUtilities.java

```diff
@@ -402,4 +404,50 @@ public static String[] getSuperTypes(String markerType) {
 			fgMarkerTypeHierarchy= new MarkerTypeHierarchy();
 		return fgMarkerTypeHierarchy.getSuperTypes(markerType);
 	}
+
+	/**
+	 * Changes the given attribute key-value pairs of the map on this marker. The values must be
+	 * <code>null</code> or an instance of one of the following classes: <code>String</code>,
+	 * <code>Integer</code>, or <code>Boolean</code>. If a value is <code>null</code>, the new value
+	 * of the attribute is considered to be undefined.
+	 *
+	 * <p>
+	 * The values of the attributes cannot be <code>String</code> whose UTF encoding exceeds 65535
+	 * bytes. On persistent markers this limit is enforced by an assertion.
+	 * </p>
+	 *
+	 * <p>
+	 * This method changes resources; these changes will be reported in a subsequent resource change
+	 * event, including an indication that this marker has been modified.
+	 * </p>
+	 * 
+	 * @param marker the marker
+	 * @param attributeChanges map with to be executed attribute changes
+	 * @see IMarker#setAttributes(String[], Object[])
+	 * @since 3.17
+	 */
+	public static void changeAttributes(IMarker marker, Map<String, Object> attributeChanges) {
+
```

Comment:
How about an early return to avoid possibly unnecessary calculations? 
```suggestion
if (!marker.exists()) {
   return;
}
```

You could then remove the `if` in line 445

Reply:
Early check will not help if marker is deleted later. Of course that can happen right after the check, but at least the code would be consistent with the rest of the code here that checks the merker state before changing it.